src/vcpkg/commands.add.cpp

```diff
@@ -45,22 +47,30 @@ namespace
                                  (msg::command_line),
                                  "",
                                  "'{command_line}' can only add one artifact at a time.");
-
 }
 
 namespace vcpkg::Commands
 {
     void AddCommand::perform_and_exit(const VcpkgCmdArguments& args, const VcpkgPaths& paths) const
     {
-        args.parse_arguments(AddCommandStructure);
+        (void)args.parse_arguments(AddCommandStructure);
         auto&& selector = args.command_arguments[0];
+
         if (selector == "artifact")
         {
             Checks::msg_check_exit(VCPKG_LINE_INFO,
                                    args.command_arguments.size() <= 2,
                                    msgAddArtifactOnlyOne,
                                    msg::command_line = "vcpkg add artifact");
-            std::string ce_args[] = {"add", args.command_arguments[1]};
+
+            auto artifact_name = args.command_arguments[1];
+
+            auto metrics = LockGuardPtr<Metrics>(g_metrics);
+            metrics->track_property("command_name", "add");
+            metrics->track_property("command_context", "artifact");
+            metrics->track_property("command_args", Hash::get_string_hash(artifact_name, Hash::Algorithm::Sha256));
```

Comment:
I think we need a file with the constants for "command_*" properties, since more commands will probably need to handle telemetry this way.